pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -406,7 +411,11 @@ func (s *discovery) getService(context parsingContext, pid int32) *model.Service
 		return nil
 	}
 
+	if s.shouldIgnorePid(proc.Pid) {
+		return nil
+	}
 	if s.shouldIgnoreComm(proc) {
+		s.addIgnoredPid(proc.Pid)
 		return nil
 	}
```

Comment:
there's inconsistency here, you have function `shouldIgnoreA` and `shouldIgnoreB`, but only one of them adds the `pid` to ignore list, while the other does not

Reply:
When a PID is first identified to be ignored, it is added to the list of ignored PIDs. All subsequent runs will continue to ignore this specific PID.